### PR TITLE
Feature/web api auditing

### PR DESCRIPTION
**Pull request by mardiel**

### CODE REVIEW

File: prime-dotnet-webapi/Controllers/EnrolmentsController.cs [hunk 1]
```diff
@@ -176,13 +178,13 @@ private bool BelongsToEnrollee(Enrolment enrolment)
             //if user is not ADMIN, check that user belongs to the enrolment
             if (!belongsToEnrollee)
             {
-                var PrimeUserId = PrimeUtils.PrimeUserId(User);
-                belongsToEnrollee = PrimeUserId != null
+                Guid PrimeUserId = PrimeUtils.PrimeUserId(User);
+                belongsToEnrollee = !Guid.Empty.Equals(PrimeUserId)
```
Comment (itchison) [inline prime-dotnet-webapi/Controllers/EnrolmentsController.cs]: I dislike this pattern - Is this there a better way?

File: prime-dotnet-webapi/Controllers/LookupsController.cs [hunk 2]
```diff
@@ -14,8 +11,9 @@ namespace Prime.Controllers
     [Produces("application/json")]
     [Route("api/[controller]")]
     [ApiController]
-    //User needs at least the ADMIN or ENROLMENT role to use this controller
-    //[Authorize(Roles = PrimeConstants.PRIME_ADMIN_ROLE + "," + PrimeConstants.PRIME_ENROLMENT_ROLE)]
+    // User needs at least the ADMIN or ENROLMENT role to use this controller
+    // FIXME - add this back once there are OAuth tokens
```
Comment (itchison) [inline prime-dotnet-webapi/Controllers/LookupsController.cs]: Can we communicate with the team that we're using FIXME as something to look for?

File: prime-dotnet-webapi/Models/IAuditable.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+using System;
+
+namespace Prime.Models
+{
+    public interface IAuditable
```
Comment (itchison) [inline prime-dotnet-webapi/Models/IAuditable.cs]: nice work here!

File: prime-dotnet-webapi/Models/BaseAuditable.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using Newtonsoft.Json;
+
+namespace Prime.Models
+{
+    public class BaseAuditable : IAuditable
```
Comment (itchison) [inline prime-dotnet-webapi/Models/BaseAuditable.cs]: Where do we set these values?

File: prime-dotnet-webapi/appsettings.json [hunk 5]
```diff
@@ -1,7 +1,7 @@
 {
   "Jwt": {
     "Audience": "prime-web-api",
-    "WellKnown": "https://sso-dev.pathfinder.gov.bc.ca/auth/realms/v4mbqqas/.well-known/openid-configuration"
+    "WellKnown": "http://localhost:8080/auth/realms/prime/.well-known/openid-configuration"
```
Comment (itchison) [inline prime-dotnet-webapi/appsettings.json]: Is this correct? How does this work in non localhost environments?

Comment (mardiel) [reply prime-dotnet-webapi/appsettings.json]: I changed this back to the dev instance - it is still not hooked up yet anyway.

File: prime-dotnet-webapi/Models/BaseAuditable.cs (hunk 4 above)
Comment (mardiel) [reply prime-dotnet-webapi/Models/BaseAuditable.cs]: These get set by the ApiDbContext as part of the SaveChanges and SaveChangesAsync methods.

File: prime-dotnet-webapi/Controllers/LookupsController.cs (hunk 2 above)
Comment (mardiel) [reply prime-dotnet-webapi/Controllers/LookupsController.cs]: I can change these to TODO, as a standard.

File: prime-angular-frontend/src/app/features/enrolment/shared/services/enrolment-state.service.ts [hunk 6]
```diff
@@ -192,7 +192,7 @@ export class EnrolmentStateService {
       hasCertification: [null, [FormControlValidators.requiredBoolean]],
       certifications: this.fb.array([]),
       isDeviceProvider: [null, [FormControlValidators.requiredBoolean]],
-      deviceProviderNumber: ['', []],
+      deviceProviderNumber: ['', [Validators.pattern('[0-9]+'), Validators.minLength(5), Validators.maxLength(5)]],
```
Comment (james-hollinger) [inline prime-angular-frontend/src/app/features/enrolment/shared/services/enrolment-state.service.ts]: If we can make our own validators, we should probably make an exactLenth validator of some kind. I could see this being used again.

File: prime-dotnet-webapi/Controllers/EnrolmentsController.cs (hunk 1 above)
Comment (mardiel) [reply prime-dotnet-webapi/Controllers/EnrolmentsController.cs]: I changed the order of the check to make this a little easier to read.  The issue is that Guid cannot be null, so they use a special Guid.Empty to represent that.  I also added some comments to the code to help explain it better.

File: prime-angular-frontend/src/app/features/enrolment/shared/services/enrolment-state.service.ts (hunk 6 above)
Comment (mardiel) [reply prime-angular-frontend/src/app/features/enrolment/shared/services/enrolment-state.service.ts]: Yes, this was just my first cut at this - I would think Martin can do a better job of this.

**Pull request merged**